_posts/2018-XX-XX-running-kafka-on-dedicated-nodes.md

```diff
@@ -0,0 +1,341 @@
+Or you can switch to using `preferredDuringSchedulingIgnoredDuringExecution` constrain which will schedule the pods more flexibly.
+
+# Node affinity
+
+_Not all nodes are born equal._
+It is quite common that a big Kubernetes or OpenShift cluster consists of many different types of nodes.
+Some are optimized for CPU heavy workloads, some for memory while other might be optiomised for storage (fast local SSDs) or network.
+Using different nodes helps to optimize both costs and performance.
+But as a user of such heterogenius cluster you need to be able to schedule your workloads to the right node.
+
+To schedule workloads onto specific nodes, Kubernetes have a feature called [node affinity](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity).
+Node affinity allows you to create a scheduling constrain for the node on which the pod will be scheduled.
+The constrain is once again specified as label selector.
+You can either use it for the built-in node label such as `beta.kubernetes.io/instance-type`.
+Or you can use your own labels to select the right node.
+
+Node affinitiy can be also specified in the `affinity` field of our custom resources for Kafka, Zookeeper and Kafka Connect.
+The example below shows configuration which will ensure that Kafka pods will be scheduled only on nodes with label `node-type` is equal to `fast-network`.
+
+```
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    ...
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-type
+                operator: In
+                values:
+                - fast-network
+    ...
+  zookeeper:
+    ...
+```
+
+When needed, you can also combine node affinity together with pod affinity:
+
+```
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: Kafka
+metadata:
+  name: my-cluster
+spec:
+  kafka:
+    ...
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+            - matchExpressions:
+              - key: node-type
+                operator: In
+                values:
+                - fast-network
+      podAntiAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+                - key: application
+                  operator: In
+                  values:
+                    - postgresql
+                    - mongodb
+          topologyKey: "kubernetes.io/hostname"
+    ...
+  zookeeper:
+    ...
+```
+
+# Dedicated nodes
+
+The ultimate solution is to create dedicated nodes.
```

Comment:
I don't think "ultimate solution" is necessarily the right way to put this, since it depends on users circumstances whether this is really required. I think it would be better to introduce it something like:

"Pod affinity is a great way to influence pod scheduling, but it's not enough entirely prevent the noisy neighbour problem."